frontend/public/actions/k8s.ts

```diff
@@ -285,42 +290,43 @@ export const watchK8sList = (
   pollAndWatch();
 };
 
-export const setAPIGroups = (value: number) => action(ActionType.SetAPIGroups, { value });
+export const setCRDCount = (value: number) => action(ActionType.SetCRDCount, { value });
 
-export const watchAPIServices = () => (dispatch, getState) => {
-  if (getState().k8s.has('apiservices') || POLLs[apiGroups]) {
-    return;
+// Update API definitions when size of CRD list changes
+const onWatchCRDListEvent = () => (dispatch, getState) => {
+  const reduxID = makeReduxID(CustomResourceDefinitionModel, {});
+  const currentCount = getState().k8s.getIn(['RESOURCES', 'crdCount'], 0);
+  const newCount = getState().k8s.getIn([reduxID, 'data']).size;
+  if (currentCount !== newCount) {
```

Comment:
Is it possible to check the watch event for ADDED or REMOVED? Then we don't have to track the count.

Reply:
Yeah, that makes sense.